lib/resourcemerge/apps_test.go

```diff
@@ -25,29 +23,29 @@ func TestEnsureDeployment(t *testing.T) {
 			name: "different replica count",
 			existing: appsv1.Deployment{
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &replicas}},
+					Replicas: int32Pointer(2)}},
 			required: appsv1.Deployment{
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &expectedReplicas}},
+					Replicas: int32Pointer(3)}},
 
 			expectedModified: true,
 			expected: appsv1.Deployment{
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &expectedReplicas}},
+					Replicas: int32Pointer(3)}},
 		},
 		{
 			name: "same replica count",
 			existing: appsv1.Deployment{
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &replicas}},
+					Replicas: int32Pointer(2)}},
 			required: appsv1.Deployment{
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &replicas}},
```

Comment:
Heh, using `replicas` instead of `expectedRepkicas` here should have tipped us off.  Thanks for the fix.

/lgtm